front/src/common/UserSettings.tsx

```diff
@@ -63,7 +66,10 @@ export default function UserSettings() {
               id="train-schedule-version-switch"
               type={SWITCH_TYPES.switch}
               name="train-schedule-version-switch"
-              onChange={() => dispatch(switchTrainScheduleV2Activated())}
+              onChange={() => {
+                dispatch(switchTrainScheduleV2Activated());
+                dispatch(updateScenarioID(undefined));
```

Comment:
Can you also update the timetableId here, to avoid to launch a useless call to the api ? You can get updateTimetableID from useOsrdConfActions (just as updateScenarioID)
```suggestion
                dispatch(updateScenarioID(undefined));
                dispatch(updateTimetableID(undefined));
```